[Rocm]: Reduce severity of detected possible memory leak from assertion to warning

arindamroy-eng opened the pull request:
Fixes #62533.
In very rare cases, the decorator for detecting memory leak is throwing assertion, even when the test is passing, and the memory is being freed with a tiny delay. The issue is not being reproduced in internal testing, but shows up sometimes in CI environment.

Reducing the severity of such detection to warning, so as not to fail the CI tests, as the actual test is not failing, rather only the check inside the decorator is failing.

Limiting the change to ROCM only for now.


cc @jeffdaily @sunway513 @jithunnair-amd @ROCmSupport @KyleCZH

arindamroy-eng committed:
[ROCM]: Reduce severity of mem leak detection

In very rare cases, the decorator for detecting
memory leak is throwing assertion, even when
the test is passing, and the memory is being
freed with a tiny delay. The issue is not being
reproduced in internal testing, but shows up
sometimes in CI environment.

Reducing the severity of such detection to warning,
so as not to fail the CI tests, as the actual test
is not failing, rather only the check inside the
decorator is failing.

Limiting the change to ROCM only for now.

Signed-off-by: Arindam Roy <rarindam@gmail.com>

jeffdaily reviewed (approved):
LGTM but I'm not seeing our CI job reported.  ROCm used to treat these as warnings, then we removed the special treatment for ROCm, and that has re-revealed why we treated these as warnings in the first place.  The memory isn't leaked.  There is a race between the memory being free'd and the runtime reporting the memory as free.

jeffdaily commented:
@pytorchbot retest this please

jeffdaily commented:
https://ci.pytorch.org/jenkins/job/pytorch-pull-request/201926/

arindamroy-eng committed:
ROCM: Skip mem leak test for ROCM

For issue #62533, stop detecting mem leak on ROCM.
It seems that the assertion in case of ROCM is a false
positive.

Signed-off-by: arindamroy-eng <rarindam@gmail.com>

arindamroy-eng commented:
@pytorchbot retest this please

arindamroy-eng commented:
@jeffdaily commented out the test itself with skipIfRocm.
The rerun seems to have passed.
